p2p/pre2p/raintree/addrbook_utils_test.go

```diff
@@ -0,0 +1,207 @@
+package raintree
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/pokt-network/pocket/p2p/pre2p/types"
+	"github.com/pokt-network/pocket/shared/config"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+type ExpectedRainTreeNetworkConfig struct {
+	numNodes          int
+	numExpectedLevels int
+}
+
+type ExpectedRainTreeMessageTarget struct {
+	level int
+	left  string
+	right string
+}
+type ExpectedRainTreeMessageProp struct {
+	orig     byte
+	numNodes int
+	addrList string
+	targets  []ExpectedRainTreeMessageTarget
+}
+
+// IMPROVE(team): Looking into adding more tests and accounting for more edge cases.
+
+func TestRainTreeAddrBookUtilsHandleUpdate(t *testing.T) {
+	cfg := &config.Config{}
+
+	addr, err := cryptoPocket.GenerateAddress()
+	require.NoError(t, err)
+
+	testCases := []ExpectedRainTreeNetworkConfig{
+		// 0 levels
+		{1, 0}, // Just self
+		// 1 level
+		{2, 1},
+		{3, 1},
+		// 2 levels
+		{4, 2},
+		{9, 2},
+		// 3 levels
+		{10, 3},
+		{27, 3},
+		// 4 levels
+		{28, 4},
+		{81, 4},
+		// 5 levels
+		{82, 5},
+		// 10 levels
+		{59049, 10},
+		// 11 levels
+		{59050, 11},
+		// 19 levels
+		// INVESTIGATE(olshansky/team): Does not scale to 1,000,000,000 nodes
+		// {1000000000, 19},
```

Comment:
I changed the comment to:

```
		// NOTE: This does not scale to 1,000,000,000 (1B) nodes because it's too slow.
		//       However, optimizing the code to handle 1B nodes would be a very premature optimization
		//       at this stage in the project's lifecycle, so the comment is simply left to inform
		//       future readers.
```